var/spack/repos/builtin/packages/caliper/package.py

```diff
@@ -154,24 +154,23 @@ def cache_test_sources(self):
     def run_cxx_example_test(self):
         """Run stand alone test: cxx_example"""
 
-        test_dir = join_path(self.test_suite.current_test_cache_dir, 'examples', 'apps')
+        test_dir = self.test_suite.current_test_cache_dir.examples.apps
+        exe = 'cxx-example'
 
-        if not os.path.isfile(join_path(test_dir, 'cxx-example.cpp')):
-            tty.msg('Skipping caliper test: file does not exist')
+        if not os.path.isfile(join_path(test_dir, exe)):
```

Comment:
Don't you mean the source file must exist here?

```suggestion
        if not os.path.isfile(join_path(test_dir, "{0}.cpp".format(exe))):
```

Reply:
Actually, since you explicitly use the source file name at line 172, you could define a variable for it, use it at 160 and again at 172.